### PR TITLE
Update node js to v20

### LINKED ISSUE #34058: Make PrestaShop compatible with Node.js 20
As [the documentation](https://devdocs.prestashop-project.org/8/basics/installation/system-requirements/#nodejs) states:

> PrestaShop uses NPM to manage frontend dependencies and [Webpack](https://webpack.js.org/) to compile them into static assets. You only need NodeJS 14.x (version 16.x is recommended, [you can get it here](https://nodejs.org/)), and NPM will take care of it all.

PrestaShop Javascript dependencies should be updated to be compatible with [standard stable Node.js versions 18 and 20](https://nodejs.dev/en/about/releases/).

### PR DESCRIPTION
**Pull request by mattgoud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | upgrade node 16 to node 20
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Make assets and try BO/FO themes, CI tests green and UI tests green
| UI Tests          | Without cache: https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/11109866596<br/>Split build: https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/11109861267
| Fixed issue or discussion?     | Fixes #34058
| Related PRs       | https://github.com/PrestaShop/hummingbird/pull/655<br/>https://github.com/PrestaShop/classic-theme/pull/155<br/>https://github.com/PrestaShop/prestashop-ui-kit/pull/238<br/>https://github.com/PrestaShop/ga.tests.ui.pr/pull/77
| Sponsor company   | Your company or customer's name goes here (if applicable).

## Transition from node 16 to node 20

- `core`/`BO-default-theme`/`BO-new-theme` :  upgrade some packages + audit fix (reduce some security warnings) to prepare for the transition to node 20
- fix some build warnings/errors
- set node 20 (docker-compose + github workflows)
- link fork FO themes upgraded in composer (see tasks)

### Tasks
- [x] `hummingbird theme`: need validate https://github.com/PrestaShop/hummingbird/pull/655 and create a new release
- [x] `classic theme` : need validate https://github.com/PrestaShop/classic-theme/pull/155 and create a new release
- [x] need update composer packages prestashop/classic and prestashap/hummingbird (when their new versions will be released)
- [x] optional : need released [prestashop-ui-kit](https://github.com/PrestaShop/prestashop-ui-kit) to fix some warnings (see https://github.com/PrestaShop/prestashop-ui-kit/pull/238)
- [x] remove forks and update composer with themes from the original prestashop repositories

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

File: .hintrc
Comment (jolelievre) [inline .hintrc:1]: For which package is this file used? Should it be in each admin theme instead of the root folder?
It seems a bit lost here 😅

File: admin-dev/themes/new-theme/js/components/entity-search-input.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ import ComponentsMap from '@components/components-map';
 import ConfirmModal from '@components/modal';
 // @ts-ignore-next-line
 import Bloodhound from 'typeahead.js';
-import {isUndefined} from '@PSTypes/typeguard';
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/entity-search-input.ts:31]: So, is there no more `PSType` "namespace"?
It's not really a problem though, it's just out of curiosity, was it a bad practice or is it not possible to create custom ones anymore?

Comment (jolelievre) [reply admin-dev/themes/new-theme/js/components/entity-search-input.ts:31]: If so we'll probably have to update all the references in the devdoc as well

File: admin-dev/themes/new-theme/js/components/modal/modal.ts [hunk 2]
```diff
@@ -185,9 +185,10 @@ export class Modal implements ModalType {
     this.$modal.modal({
       backdrop: closable ? true : 'static',
       keyboard: closable !== undefined ? closable : true,
-      show: false,
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/modal/modal.ts:188]: This one is more a breaking change, people are used to have this option available for the init, why doesn't it work anymore? If we can keep the original behaviour it would be better

File: admin-dev/themes/new-theme/js/components/pagination/dynamic-paginator.ts [hunk 3]
```diff
@@ -23,8 +23,8 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
-import PaginationServiceType from '@js/types/services';
-import RendererType from '@js/types/renderers';
+import PaginationServiceType from '@js/@types/services';
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/pagination/dynamic-paginator.ts]: How come we need an `@` everywhere on all the components that are in `@js/types` ?
(I ask on this file but it's a valid question for all similar modifications)

File: admin-dev/themes/new-theme/js/components/showcase-card/extension/showcase-card-close-extension.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ export default class ShowcaseCardCloseExtension {
    */
   extend(helperBlock: ShowcaseCard): void {
     const container = helperBlock.getContainer();
-    container.on('click', '.js-remove-helper-block', (evt) => {
+    container.on('click', '.js-remove-helper-block', (evt: { target: any; }) => {
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/showcase-card/extension/showcase-card-close-extension.ts]:
```suggestion
    container.on('click', '.js-remove-helper-block', (evt: Event) => {
```
Maybe not the right type, but isn't there a native type for events that could be used here?

File: admin-dev/themes/new-theme/template/header.tpl [hunk 5]
```diff
@@ -24,7 +24,7 @@
  *}
 <meta charset="utf-8">
 <meta name="viewport" content="width=device-width, initial-scale={(isset($viewport_scale)) ? $viewport_scale : '1'}">
-<meta name="apple-mobile-web-app-capable" content="yes">
+<meta name="mobile-web-app-capable" content="yes">
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/template/header.tpl:27]: Again out of curiosity, why is this changed needed, and what is its impact?

File: admin-dev/themes/new-theme/tsconfig.json [hunk 6]
```diff
@@ -20,9 +21,8 @@
       "@scss/*": ["scss/*"],
       "@node_modules/*": ["node_modules/*"],
       "@PSVue/*": ["js/vue/*"],
-      "@PSTypes/*": ["js/types/*"]
+      "@PSTypes/*": ["js/@types/*"],
```
Comment (jolelievre) [inline admin-dev/themes/new-theme/tsconfig.json:24]: Ah seems like the "namespace" (I'm not sure it's the right name for it) still exists, so why did you need to change it in all the files regardless?

File: tools/assets/build.sh [hunk 7]
```diff
@@ -30,6 +30,7 @@ function build {
   fi
 
   touch buildLock
+  chmod 664 buildLock
```
Comment (jolelievre) [inline tools/assets/build.sh:33]: Ah, why did you need this? A problem of permission depending on which user you use to lunch the asset building maybe?

File: .hintrc
Comment (mattgoud) [reply .hintrc:1]: I will gitignore

File: admin-dev/themes/new-theme/js/components/entity-search-input.ts (hunk 1 above)
Comment (mattgoud) [reply admin-dev/themes/new-theme/js/components/entity-search-input.ts:31]: ok done !

File: admin-dev/themes/new-theme/js/components/showcase-card/extension/showcase-card-close-extension.ts (hunk 4 above)
Comment (mattgoud) [reply admin-dev/themes/new-theme/js/components/showcase-card/extension/showcase-card-close-extension.ts]: ok done, replace with `JQuery.Event`

File: admin-dev/themes/new-theme/js/components/modal/modal.ts (hunk 2 above)
Comment (mattgoud) [reply admin-dev/themes/new-theme/js/components/modal/modal.ts:188]: for fix typescript error :
`this.$modal.modal('hide')` instead `show: false`

**Review by jolelievre — APPROVED** [review]
Thanks @mattgoud

**Review by boherm — APPROVED** [review]
